Add LocalizerDirection op mode to debug odometry hardware installation

monkbroc opened the pull request:
Road Runner has an op mode for debugging odometry hardware installation. The instructions for Pedro Pathing say to make sure the encoder ticks are increasing when the robot goes forward and right, but there's no easy way to check that.

This PR adds the LocalizerDirection op mode and the `debug` method to `Localizer` to show the raw output of the sensors used by the localizer.

monkbroc committed:
Fix dashboard URL

monkbroc committed:
Add LocalizerDirection op mode to debug odometry hardware installation

monkbroc committed:
Add instructions to check motor names and reverse when using LocalizationTest

BaronClaps commented:
You can do this with LocalizationTest?

monkbroc commented:
LocalizationTest is for confirming that localization is good at the end of the process. LocalizerDirection is for ensuring that sensors output the correct values at the start of the process.

AnyiLin commented:
I dont see this adding anything significant that localization test already doesnt do

monkbroc commented:
Here's our experience: when setting up localization with 3 dead wheel encoders, the instructions say "reverse the direction of any encoders so that the forward encoders tick up when the robot is moving forward and the strafe encoder ticks up when the robot moves right." There is no op mode to see the encoder ticks. We assumed that the default config was ok and moved on with Turn Localizer Tuner. The computed angle was changing wildly so I guessed that some encoders were backwards but I wasn't sure which one. So we added this op mode to see the value of all encoders and make sure that they increase in the correct direction.

Same thing with the IMU. There's no op mode to check that you've done the step "change the orientation of the IMU to match that of your robot's" correctly.

Team that use the library can implement their own op modes for debugging the installation of their odometry sensors, but including one in the quickstart seems useful to me. If you don't think this is useful, feel free to close the PR.

Logan-Nash commented:
I will rewrite part of the instructions in order to clarify the odometry tuning process. I see no real value of adding in such a process but I do agree that the wording is ambiguous and will be changed in the ReadMes and the website in order to reflect that. Thanks for the suggestion!

monkbroc commented:
Sounds good. Thanks for considering the feedback.
